hack/lib/util/environment.sh

```diff
@@ -0,0 +1,253 @@
+#!/bin/bash
+
+# This script holds library functions for setting up the shell environment for OpenShift scripts
+#
+# This script assumes $OS_ROOT is set before being sourced
+source "${OS_ROOT}/hack/util.sh"
+
+# os::util::environment::use_sudo updates $USE_SUDO to be 'true', so that later scripts choosing between
+# execution using 'sudo' and execution without it chose to use 'sudo'
+#
+# Globals:
+#  None
+# Arguments:
+#  None
+# Returns:
+#  - export USE_SUDO
+function os::util::environment::use_sudo() {
+    export USE_SUDO=true
+}
+
+# os::util::environment::setup_time_vars sets up environment variables that describe durations of time
+# These variables can be used to specify times for other utility functions
+#
+# Globals:
+#  None
+# Arguments:
+#  None
+# Returns:
+#  - export TIME_MS
+#  - export TIME_SEC
+#  - export TIME_MIN
+function os::util::environment::setup_time_vars() {
+    export TIME_MS=1
+    export TIME_SEC="$(( 1000  * ${TIME_MS} ))"
+    export TIME_MIN="$(( 60 * ${TIME_SEC} ))"
+}
+
+# os::util::environment::setup_all_server_vars sets up all environment variables necessary to configure and start an OpenShift server
+#
+# Globals:
+#  - OS_ROOT
+#  - PATH
+#  - TMPDIR
+#  - LOG_DIR
+#  - ARTIFACT_DIR
+#  - KUBELET_SCHEME
+#  - KUBELET_BIND_HOST
+#  - KUBELET_HOST
+#  - KUBELET_PORT
+#  - BASE_TMPDIR
+#  - ETCD_PORT
+#  - ETCD_PEER_PORT
+#  - API_BIND_HOST
+#  - API_HOST
+#  - API_PORT
+#  - API_SCHEME
+#  - PUBLIC_MASTER_HOST
+#  - USE_IMAGES
+# Arguments:
+#  - 1: the path under the root temporary directory for OpenShift where these subdirectories should be made
+# Returns:
```

Comment:
Depends on how lenient reviews are, doesn't it? We could say the same thing about Godoc, and we're about to base our Swagger API descriptions on Godoc, even.